### PR TITLE
Method refactor test chexpert

### PR DESCRIPTION
**Pull request by abarancikova**
Add chexpert (only effusion)

### CODE REVIEW

File: runs/CheXpert/get_chexpert_config.py
Comment (CynthiaFeeney) [inline runs/CheXpert/get_chexpert_config.py:1]: Is this different from the other dataset config code? If so, add some comments to reflect that and move this file to the same folder as the more general dataset config code. If not, delete this file.

File: runs/CheXpert/only_atelectasis.py
Comment (CynthiaFeeney) [inline runs/CheXpert/only_atelectasis.py:1]: Should put this within src.

File: runs/CheXpert/only_effusion.py
Comment (CynthiaFeeney) [inline runs/CheXpert/only_effusion.py:1]: Should put this within src.

File: src/ssl_bench/config.py
Comment (CynthiaFeeney) [inline src/ssl_bench/config.py:1]: Could we switch the names to be more descriptive? "CheXpertEffusion" for example?

File: src/ssl_bench/dataset_csv.py [hunk 1]
```diff
@@ -100,10 +100,15 @@ def __init__(self, csv_file, root_dir, transforms=None, transform=None,
         self.data = pd.read_csv(csv_file)
         self.root_dir = root_dir
         self.transform = transform
+        # self.label_columns = ['Edema', 'Consolidation',
```
Comment (CynthiaFeeney) [inline src/ssl_bench/dataset_csv.py]: Remove commented code.

File: src/ssl_bench/hyper_search.py [hunk 2]
```diff
@@ -407,8 +409,32 @@ def train_one_epoch(args, model, optimizer, scheduler, train_loader, epoch):
     all_logits = torch.cat(all_logits)
     all_labels = torch.cat(all_labels)
 
+    # for epoch in range(args.start_epoch, args.train_epoch):
```
Comment (CynthiaFeeney) [inline src/ssl_bench/hyper_search.py]: Remove commented code.

File: src/ssl_bench/hyper_search.py [hunk 3]
```diff
@@ -407,8 +409,32 @@ def train_one_epoch(args, model, optimizer, scheduler, train_loader, epoch):
...
+    #         args, model, optimizer, scheduler, train_loader, epoch)
+    #     pr.disable()
+    #     pr.dump_stats("latest.stats")
+    #     stats = Stats(pr)
+    #     stats.sort_stats("tottime").print_stats(25)
+    #     pr.print_stats(sort='time')
+
     return labeled_loss.avg, all_logits, all_labels
 
+def plot_confusion_matrix(cm, class_names):
```
Comment (CynthiaFeeney) [inline src/ssl_bench/hyper_search.py]: Great function, should probably be in a different file like `src/ssl_bench/utils/eval_utils.py`.

File: src/ssl_bench/methods/LabelOnlyBaseline.py [hunk 4]
```diff
@@ -9,7 +9,14 @@ def forward(self, l_data, l_labels):
         logits = self.backbone(l_data)
         class_probs = func.softmax(logits, dim=1)
 
-        s_loss = func.cross_entropy(logits, l_labels, weight=self.args.weights, reduction='mean')
+        if self.args.dataset_name == "CheXpert":
```
Comment (CynthiaFeeney) [inline src/ssl_bench/methods/LabelOnlyBaseline.py]: Instead of making it dataset specific, couldn't we check that there are only logits for one class to make things more general?

File: src/ssl_bench/methods/LabelOnlyBaseline.py [hunk 5]
```diff
@@ -9,7 +9,14 @@ def forward(self, l_data, l_labels):
...
+            s_loss = func.binary_cross_entropy_with_logits(
+                logits, l_labels.float(), weight=self.args.weights, reduction='mean')
+        else:
+            s_loss = func.cross_entropy(
+                logits, l_labels, weight=self.args.weights, reduction='mean')
+
+        # s_loss = func.cross_entropy(logits, l_labels, weight=self.args.weights, reduction='mean')
```
Comment (CynthiaFeeney) [inline src/ssl_bench/methods/LabelOnlyBaseline.py]: Remove commented code.

**Review by CynthiaFeeney — CHANGES_REQUESTED** [review]

**Review by CynthiaFeeney — APPROVED** [review]
